portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -85,6 +87,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
      */
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        UserService userService = UserServiceFactory.getUserService();
+        if (!userService.isUserLoggedIn()) {
+            response.sendRedirect("/#comments");
+            return;
+        }
+
+        String userEmail = userService.getCurrentUser().getEmail();
         String username = request.getParameter("name");
```

Comment:
Let's make all the strings here private static final. For example:
private final static PARAM_NAME = "name";